### ISSUE TITLE
[Example Request] NVIDIA DALI LightningDataModule

### ISSUE THREAD

**Issue by austinmw** [airctic/icevision#1067]
## 📓 New <Tutorial/Example>
**Is this a request for a tutorial or for an example?**  
Either really.

**What is the task?**
Object detection

**Is this example for a specific model?**
Any detection model

**Is this example for a specific dataset?**
Any LightningDataModule that uses nvidia DALI. Maybe COCO?

I have a LightningDataModule that uses NVIDIA DALI. The train and val dataloaders return a single item, `batch`, which is a dictionary that contains the following items:

`batch['images']` - a batch of preprocessed tensor images, for example of size `(batch_size, 3, 416, 416)`
`batch['bboxes']` - a padded tensor of bboxes, with format `x,y,x2,y2`, for example of size `(batch_size, 200, 4)`
`batch['labels']` - a padded tensor of class labels, for example of size `(batch_size, 200)`

I'm not sure how to overwrite `training_step` and `validation_step` to incorporate a dali datamodule like this. Especially, what is `records` in `(xb, yb), records = batch`, and what do I need to do with `self.convert_raw_predictions`?
 

---

**Don't remove**
Main issue for examples: #39
